packages/auth/src/core/strategies/passkey.ts

```diff
@@ -0,0 +1,244 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Auth, User, UserCredential } from '../../model/public_types';
+
+import {
+  startPasskeyEnrollment,
+  StartPasskeyEnrollmentRequest,
+  StartPasskeyEnrollmentResponse,
+  finalizePasskeyEnrollment,
+  FinalizePasskeyEnrollmentRequest,
+  FinalizePasskeyEnrollmentResponse,
+  startPasskeySignIn,
+  StartPasskeySignInRequest,
+  StartPasskeySignInResponse,
+  finalizePasskeySignIn,
+  FinalizePasskeySignInRequest,
+  FinalizePasskeySignInResponse,
+  publicKeyCredentialToJSON
+} from '../../api/account_management/passkey';
+import { UserInternal } from '../../model/user';
+import { _castAuth } from '../auth/auth_impl';
+import { getModularInstance } from '@firebase/util';
+import { OperationType } from '../../model/enums';
+import { UserCredentialImpl } from '../user/user_credential_impl';
+import { signInAnonymously } from './anonymous';
+
+const DEFAULT_PASSKEY_ACCOUNT_NAME = 'Unnamed account (Web)';
+const PASSKEY_LOOK_UP_ERROR_MESSAGE =
+  'The operation either timed out or was not allowed.';
+
+/**
+ * Signs in a user with a passkey.
+ * @param auth - The Firebase Auth instance.
+ * @param name - The user's name for passkey.
+ * @param manualSignUp - Creates an anonymous user if a user does not exist. Defaults to false.
+ * @returns A promise that resolves with a `UserCredential` object.
+ */
+export async function signInWithPasskey(
+  auth: Auth,
+  name: string,
+  manualSignUp: boolean = false
+): Promise<UserCredential> {
+  const authInternal = _castAuth(auth);
+
+  // Start Passkey Sign in
+  const startSignInRequest: StartPasskeySignInRequest = {};
+  const startSignInResponse = await startPasskeySignIn(
+    authInternal,
+    startSignInRequest
+  );
+
+  const options = getPasskeyCredentialRequestOptions(startSignInResponse, name);
+
+  // Get the credential
+  let credential;
+  try {
+    credential = (await navigator.credentials.get({
+      publicKey: options
+    })) as PublicKeyCredential;
+
+    const finalizeSignInRequest: FinalizePasskeySignInRequest = {
+      authenticatorAuthenticationResponse:
+        publicKeyCredentialToJSON(credential),
+      name,
+      displayName: name
+    };
+    const finalizeSignInResponse: FinalizePasskeySignInResponse =
+      await finalizePasskeySignIn(authInternal, finalizeSignInRequest);
+
+    const operationType = OperationType.SIGN_IN;
+    const userCredential = await UserCredentialImpl._fromIdTokenResponse(
+      authInternal,
+      operationType,
+      finalizeSignInResponse
+    );
+    await auth.updateCurrentUser(userCredential.user);
+    return userCredential;
+  } catch (error) {
+    if (
+      (error as Error).message.includes(PASSKEY_LOOK_UP_ERROR_MESSAGE) &&
+      !manualSignUp
```

Comment:
Should this be "manualSignUp" instead of !manualSignUp?

Reply:
Nope. Note that it's Manual(instead of Auto) sign up, so we create the user when it's false.

Reply:
Nope. Note that it's Manual(instead of Auto) sign up, so we create the user when it's false.